packages/nx/src/project-graph/project-graph-builder.ts

```diff
@@ -268,7 +274,8 @@ export class ProjectGraphBuilder {
     }
     if (
       !this.graph.nodes[targetProjectName] &&
-      !this.graph.externalNodes[targetProjectName]
+      !this.graph.externalNodes[targetProjectName] &&
+      !sourceProjectFile
```

Comment:
Do not break if adding explicit dependency to a file could not be found. We will filter it later in the `getUpdatedProjectGraph` if not resolved by then

Reply:
This makes sense for this use case. I'm a bit concerned that this would make it somewhat confusing for legitimate mistakes without any feedback.

If dependencies were drawn to projects that did not exist and will never exist because of some legitimate issue, the plugin author would not be told that the dependency was incorrect right? They would just wonder why they don't see it in the graph?

I suppose we could add a flag or a new method which would allow plugin authors to be explicit that they are adding a dependency which they know does not exist... But I'm hesitant to expand the API.

Reply:
Please note that `move` is broken - tracking it in a separate issue https://github.com/nrwl/nx/issues/16879